benchmark/run_all_benchmarks.sh

```diff
@@ -0,0 +1,64 @@
+EXECUTOR=$1
+
+SSGET=ssget
+NUM_PROBLEMS=$(${SSGET} -n)
+
+
+keep_latest() {
+    RESULT=$1
+    for file in $@; do
+        if [ ${file} -nt ${RESULT} ]; then
+            RESULT=${file}
+        fi
+    done
+    if [ "${RESULT}" != "$1" ]; then
+        cp ${RESULT} $1
+    fi
+    for file in ${@:2}; do
+        rm -f ${file}
+    done
+}
+
+
+for (( i=1; i <= ${NUM_PROBLEMS}; ++i )); do
+    if [ $(${SSGET} -i $i -preal) = "0" ]; then
+        ${SSGET} -i $i -c >/dev/null
+        continue
+    fi
+    RESULT_DIR="results/${SYSTEM_NAME}/${EXECUTOR}/SuiteSparse"
+    GROUP=$(${SSGET} -i $i -pgroup)
+    NAME=$(${SSGET} -i $i -pname)
+    RESULT_FILE="${RESULT_DIR}/${GROUP}/${NAME}.json"
+    mkdir -p $(dirname ${RESULT_FILE})
+    INPUT=$(${SSGET} -i $i -e)
+    echo "[{\"filename\": \"${INPUT}\", \"problem\": $(${SSGET} -i $i -j)}]" \
+        >"${RESULT_FILE}.imd"
+    echo -e "($i/${NUM_PROBLEMS}):\tRunning SpMV for ${GROUP}/${NAME}" 1>&2
+    ./spmv/spmv --executor=${EXECUTOR} \
+                --backup="${RESULT_FILE}.bkp" \
```

Comment:
please explain here what is happening: write results into several files (backup) etc....